### PR TITLE
HTML imports

### PR DESCRIPTION
**Pull request by tapina**
IMD and HTML imports rather than System JS and .js/.css files.

Components, helpers, examples and tests migrated.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,8 +4,12 @@
     "amd": true,
     "browser": true
   },
+  "plugins": [
+    "html"
+  ],
   "globals": {
-    "ActiveXObject": true
+    "ActiveXObject": true,
+    "SystemJS": true
```
Comment (kevmarchant) [inline .eslintrc]: don't need systemjs to be global here now, do we?

File: components/ca-barcode-scanner.html [hunk 2]
```diff
@@ -1,11 +1,13 @@
+<link rel="import" href="./helpers/create-element.html">
+<script>
 /* global cordova */
-define(['./helpers/create-element.js', 'document-register-element'], (createElement) => {
+define(['create-element'], (createElement) => {
```
Comment (kevmarchant) [inline components/ca-barcode-scanner.html]: don't need brackets around createElement

File: components/ca-barcode-scanner.html [hunk 3]
```diff
@@ -34,8 +36,11 @@
          */
         attributeChangedCallback(attrName, oldVal, newVal) {
             switch (attrName) {
-                case 'title': this.button.textContent = newVal; break;
-                default: break;
+                case 'title':
```
Comment (kevmarchant) [inline components/ca-barcode-scanner.html]: this whole lot can be one if statement

File: components/ca-currency.html [hunk 4]
```diff
@@ -1,9 +1,39 @@
...
+        content: attr(symbol);
+        line-height: 1;
+        text-align: center;
+        background: #eee;
+        border: 1px solid #ccc;
+        border-radius: 4px 0px 0px 4px;
+    }
+</style>
+<script>
+define(['create-element'], (createElement) => {
```
Comment (kevmarchant) [inline components/ca-currency.html]: unnecessary brackets (won't keep pointing these out)

File: components/ca-form-builder.html [hunk 5]
```diff
@@ -1375,3 +1607,4 @@
     // Register our new element
     document.registerElement('ca-form-builder', FormBuilder);
```
Comment (kevmarchant) [inline components/ca-form-builder.html]: Should now be CaFormBuilderElement

File: components/ca-currency.html (hunk 4 above)
Comment (jagreehal) [inline components/ca-currency.html]: @kevmarchant keep pointing these out... maybe add a lint rule?

**Pull request merged**